crypto/bio/bio_meth.c

```diff
@@ -63,12 +63,45 @@ int BIO_meth_set_write(BIO_METHOD *biom,
 
 int (*BIO_meth_get_read(BIO_METHOD *biom)) (BIO *, char *, int)
 {
+    return biom->bread_old;
+}
+
+int (*BIO_meth_get_read_ex(BIO_METHOD *biom)) (BIO *, char *, size_t, size_t *)
+{
     return biom->bread;
 }
 
+/* Conversion for old style bread to new style */
+int bread_conv(BIO *bio, char *out, size_t outl, size_t *read)
+{
+    int ret;
+
+    if (outl > INT_MAX)
+        return 0;
+
+    ret = bio->method->bread_old(bio, out, (int)outl);
+
+    if (ret <= 0) {
+        *read = 0;
+        return ret;
+    }
```

Comment:
Doesn't `return == 0` mean "there was zero bytes of data available to read at this moment / location"? I wouldn't necessarily call that a failure, and most definitely not on a non-smoking BIO.